### PR TITLE
Dev documentation + setup

### PR DESCRIPTION
**Pull request by reinout**

### CODE REVIEW

**Comment by reinout** [comment]
Biggest change (though small) is that I `-e .` install ourselves in the requirements instead of separately. Less duplication that way.

And I added some developer documentation on how to install/run it locally. For that I also moved some tool settings to `setup.cfg`.

**Comment by reinout** [comment]
And, final comment, I didn't have a python 3.7 handy, so the requirements.txt misses pins for a couple of python 3.7-only dependencies. This isn't really a problem, though, I guess.

**Comment by gforcada** [comment]
We can do full `black` without single quotes, i did copy and paste the configuration I had on other projects I have, and forgot to fix that, and as you mentioned on the other PR, I was about to do it, and then forgot about it completely again 😅

File: README.rst [hunk 1]
```diff
@@ -123,3 +123,30 @@ requests in the github issue tracker.
...
+
+  $ bin/pip-compile requirements.in
+
+To run the tests (there's some pytest configuration in ``setup.cfg``)::
+
+  $ bin/pytest
+
+Some checks that are run by github actions::
+
+  $ bin/black --skip-string-normalization
```
Comment (gforcada) [inline README.rst:147]: As said, we can remove this `--skip-string-normalization` and use double quotes

File: README.rst [hunk 2]
```diff
@@ -123,3 +123,30 @@ requests in the github issue tracker.
...
+  $ bin/codespell setup.py z3c/
+  $ bin/flake8 setup.py z3c/
+
+Note that the string quoting style is currently not black-standard, so you'll
```
Comment (gforcada) [inline README.rst:151]: We can remove this then 😄

File: requirements.in [hunk 3]
```diff
@@ -1,7 +1,7 @@
-stdlib-list
-cached-property
-toml
+# Ourselves, including our dependencies
+-e file:.#z3c.dependencychecker[test]
```
Comment (gforcada) [inline requirements.in:2]: I did not know that you could do it this way, I was always using the shorthand `-e .[test]` 😄 

Is there any benefit on stating this `file:.#z3c.dependencychecker` ? 🤔

File: requirements.txt [hunk 4]
```diff
@@ -1,23 +1,29 @@
 #
-# This file is autogenerated by pip-compile with python 3.7
+# This file is autogenerated by pip-compile with python 3.8
```
Comment (gforcada) [inline requirements.txt:2]: I have python3.7 locally, should I run it and add a commit on top?

File: z3c/dependencychecker/tests/test_user_mappings.py [hunk 5]
```diff
@@ -132,6 +132,30 @@ def test_one_user_mapping(minimal_structure):
     } == mappings
 
 
+def test_filter_out_mappings_on_test(minimal_structure):
```
Comment (gforcada) [inline z3c/dependencychecker/tests/test_user_mappings.py:135]: nice, indeed your test covers the code that was missing ✨

**Comment by reinout** [comment]
Let's do the quotes in a separate pull request :-)

**Review by gforcada — APPROVED** [review]
Great job!

File: requirements.in (hunk 3 above)
Comment (reinout) [reply requirements.in:2]: I think it was for some other tool... That way, the installation mechanism knows beforehand which package it is going to be dev-installing. I might remember the reason in a few weeks.

File: requirements.txt (hunk 4 above)
Comment (reinout) [reply requirements.txt:2]: Fine. Or perhaps easier to do it in one of the other commits afterwards. Let me first juggle the various pull requests :-)

**Comment by reinout** [comment]
Ok, I'm merging this, then you can finish off #96 (including the 3.7 trick). And afterwards we can do a separate PR for the quotes.

**Comment by reinout** [comment]
Oh, and thanks for putting the `pyupgrade` tool in there. I didn't know it existed and it is already saving me quite some time!

**Pull request merged**